core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -746,10 +755,11 @@ private boolean refreshIncremental0(
 
     private void resetInvalidState(MatViewState state, WalWriter walWriter) {
         state.markAsValid();
+        state.setLastRefreshBaseTableTxn(-1);
```

Comment:
why is this reset to -1?
would not this trigger a huge incremental (equivalent to full) refresh next time the base table is updated?

Reply:
We weren't resetting the txn before in full refresh relying on the txn assignment done by `insertAsSelect()`. Since now we `resetInvalidState()` before the `insertAsSelect()` call, it's safer to reset the txn here. Say, if the process crashes after the `resetInvalidState()` call, the next incremental refresh won't pick up old txn, but instead it'll act as a full refresh which is correct.

Reply:
maybe my version of this change was better then?
so we could run `resetInvalidState()` only after a successful `insertAsSelect()`, otherwise the view stays invalid.

Reply:
ok, fine.